src/main/java/com/google/devtools/build/lib/skyframe/SkyframeExecutor.java

```diff
@@ -1130,6 +1161,10 @@ public Root getForcedSingleSourceRootIfNoExecrootSymlinkCreation() {
     return null;
   }
 
+  final boolean externalRepositoriesEnabled() {
+    return shouldStoreTransitivePackagesInLoadingAndAnalysis();
```

Comment:
Please invert the call: have `shouldStoreTransitivePackagesInLoadingAndAnalysis` call `externalRepositoriesEnabled`.